workflows/dnaseq/dnaseq-core.wdl

```diff
@@ -0,0 +1,160 @@
+## **WARNING:** this workflow is experimental! Use at your own risk!
+
+version 1.1
+
+import "../../data_structures/read_group.wdl"
+import "../../tools/bwa.wdl"
+import "../../tools/picard.wdl"
+import "../../tools/samtools.wdl"
+import "../../tools/util.wdl"
+import "../general/bam-to-fastqs.wdl" as bam_to_fastqs_wf
+import "../general/samtools-merge.wdl" as samtools_merge_wf
+
+workflow dnaseq_core_experimental {
+    meta {
+        description: "Aligns DNA reads using bwa"
+        outputs: {
+            harmonized_bam: "Harmonized DNA-Seq BAM, aligned with bwa",
+            harmonized_bam_index: "Index for the harmonized DNA-Seq BAM file"
+        }
+        allowNestedInputs: true
+    }
+    parameter_meta {
+        read_one_fastqs_gz: "Input gzipped FASTQ format file(s) with 1st read in pair to align"
+        read_two_fastqs_gz: "Input gzipped FASTQ format file(s) with 2nd read in pair to align"
+        bwa_db: "Gzipped tar archive of the bwa reference files. Files should be at the root of the archive."
+        reads_per_file: "Number of reads per FASTQ file to output."
+        read_groups: "A string containing the read group information to output in the BAM file. If including multiple read group fields per-read group, they should be space delimited. Read groups should be comma separated, with a space on each side (i.e. ' , '). The ID field must come first for each read group and must be contained in the basename of a FASTQ file or pair of FASTQ files if Paired-End. Example: `ID:rg1 PU:flowcell1.lane1 SM:sample1 PL:illumina LB:sample1_lib1 , ID:rg2 PU:flowcell1.lane2 SM:sample1 PL:illumina LB:sample1_lib1`. These two read groups could be associated with the following four FASTQs: `sample1.rg1_R1.fastq,sample1.rg2_R1.fastq` and `sample1.rg1_R2.fastq,sample1.rg2_R2.fastq`"
+        prefix: "Prefix for the BAM file. The extension `.bam` will be added."
+        aligner: {
+            description: "BWA aligner to use",
+            choices: ["mem", "aln"]
+        }
+        validate_input: "Ensure input BAM is well-formed before beginning harmonization?"
+        use_all_cores: "Use all cores? Recommended for cloud environments."
+    }
+    input {
+        Array[File] read_one_fastqs_gz
+        Array[File] read_two_fastqs_gz
+        File bwa_db
+        Int reads_per_file = 10000000
+        Array[ReadGroup] read_groups
+        String prefix
+        String aligner = "mem"
+        Boolean use_all_cores = false
+    }
+
+    call parse_input { input:
+        aligner=aligner
+    }
+
+    scatter (rg in read_groups) {
+        call read_group.ReadGroup_to_string { input: read_group=rg }
+    }
+
+    Array[String] read_groups_bwa = prefix("@RG\t", ReadGroup_to_string.stringified_read_group)
```

Comment:
There are other instances of `"\t"` elsewhere in this PR. I'm confused by how these escape sequences are being interpreted.

Reply:
We could probably rethink this. The `sub()` commands below were from your original pipeline. At the time, we were taking `@RG` records directly from the BAM header, so there was a literal tab in the string. So the logic below was replacing that with the appropriate escapes. Here, since we accept the `ReadGroup` struct instead, I am just putting literal tabs in, which admittedly are replaced a few lines below.

Though I'm now looking at the implementation of `ReadGroup_to_string` and seeing that the fields are space-delimited. So this may not work as intended.

Reply:
Previously this was using a `\t` between `@RG` and the fields and then spaces. The `bwa` docs state `\t` for all fields, but it was working (feature or bug? I don't know). I've updated this to use the tab between all fields.